client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/test/checklist-item.tsx

```diff
@@ -44,6 +44,38 @@ describe( 'ChecklistItem', () => {
 			const taskButton = screen.queryByRole( 'link' );
 			expect( taskButton ).toHaveAttribute( 'disabled' );
 		} );
+
+		describe( 'and the task is kept active', () => {
+			it( 'enables the task', () => {
+				render( <ChecklistItem task={ getTask( { isCompleted: true, keepActive: true } ) } /> );
+				const taskButton = screen.queryByRole( 'link' );
+				expect( taskButton ).not.toHaveAttribute( 'disabled' );
+			} );
+
+			it( 'hides the task enabled icon', () => {
+				render( <ChecklistItem task={ getTask( { isCompleted: true, keepActive: true } ) } /> );
+				const taskEnabledIcon = screen.queryByLabelText( 'Task enabled' );
+				expect( taskEnabledIcon ).toBeFalsy();
+			} );
+		} );
+	} );
+
+	describe( 'when a task is incomplete', () => {
+		it( 'hides the task complete icon', () => {
+			render( <ChecklistItem task={ getTask( { isCompleted: false } ) } /> );
+			const taskCompleteIcon = screen.queryByLabelText( 'Task complete' );
+			expect( taskCompleteIcon ).toBeFalsy();
+		} );
+		it( 'shows the task enabled icon', () => {
+			render( <ChecklistItem task={ getTask( { isCompleted: false } ) } /> );
+			const taskEnabledIcon = screen.queryByLabelText( 'Task enabled' );
+			expect( taskEnabledIcon ).toBeTruthy();
+		} );
+		it( 'enables the task', () => {
```

Comment:
Im slightly confused reading through these test descriptions.

* "when a task is incomplete -> shows the task enabled icon" (same with "enabled the task")- doesn't always seem to be a case to assert, since some incomplete tasks are disabled (like "Launch" LIB before the "add links" dependency is met)

Similarly above (not changes made in this PR) we have "when the task is complete -> disables the task" - which doesn't always seem true since we have completed tasks that are still enabled?

Reply:
> Im slightly confused reading through these test descriptions.

> "when a task is incomplete -> shows the task enabled icon" (same with "enabled the task")- doesn't always seem to be a case to assert, since some incomplete tasks are disabled (like "Launch" LIB before the "add links" dependency is met)

> Similarly above (not changes made in this PR) we have "when the task is complete -> disables the task" - which doesn't always seem true since we have completed tasks that are still enabled?

Good observations -- I'll amend the tests to reflect this

Reply:
> Similarly above (not changes made in this PR) we have "when the task is complete -> disables the task" - which doesn't always seem true since we have completed tasks that are still enabled?

This case was already covered by the tests I introduced

https://github.com/Automattic/wp-calypso/pull/68053/files#diff-c1574bb61f7c32711cb6f8e19242c2af6546d41f576e8894644783599c013135R48-R61

Reply:
> "when a task is incomplete -> shows the task enabled icon" (same with "enabled the task")- doesn't always seem to be a case to assert, since some incomplete tasks are disabled (like "Launch" LIB before the "add links" dependency is met)

Handled in [4344d9b](https://github.com/Automattic/wp-calypso/pull/68053/commits/4344d9b1af04b09cf25f954ba193bff8beb97a08)